source/agora/consensus/data/Block.d

```diff
@@ -843,3 +847,55 @@ unittest
     assert(merkle_path[3] == habcdefgh);
     assert(block.header.merkle_root == Block.checkMerklePath(hi, merkle_path, 8));
 }
+
+/// demonstrate signing two blocks at height 1 to reveal private node key
+unittest
+{
+    import agora.consensus.data.genesis.Test: GenesisBlock;
+    import agora.crypto.ECC: Scalar, Point;
+    import agora.crypto.Schnorr;
+    import agora.utils.Test;
+    import std.format;
+
+    Hash random_seed1 = "seed1".hashFull();
+    Hash random_seed2 = "seed2".hashFull();
+    Hash preimage = "preimage".hashFull();
+
+    // Generate two blocks at height 1
+    auto block1 = GenesisBlock.makeNewBlock(
+        genesisSpendable().take(2).map!(txb => txb.sign()).array(), 10, random_seed1, 6);
+    auto block2 = GenesisBlock.makeNewBlock(
+        genesisSpendable().take(3).map!(txb => txb.sign()).array(), 10, random_seed2, 6);
```

Comment:
Why is the call to `array` used ? Also can we name those constants ?

Reply:
Updated